lib/SMTLIB2/Solver.cpp

```diff
@@ -133,6 +152,42 @@ struct SMTLIBParser {
 
     return APInt(Width, StringRef(NumBegin, NumEnd - NumBegin), 10);
   }
+
+  APInt parseHexBitVector(std::string &ErrStr) {
+    ErrStr.clear();
+    const char *NumBegin = Begin;
+    while (Begin != End && ((*Begin >= '0' && *Begin <= '9') ||
+           (*Begin >= 'a' && *Begin <= 'f') ||
+           (*Begin >= 'A' && *Begin <= 'F')))
+      ++Begin;
+    const char *NumEnd = Begin;
+
+    if (!consumeExpected(')', ErrStr))
+      return APInt();
+    if (!consumeExpected(')', ErrStr))
+      return APInt();
```

Comment:
The above four lines now seem duplicated in the three routines to parse literals. Should these occur in parseModel instead?